WordPress/src/main/java/org/wordpress/android/ui/sitecreation/SearchInputWithHeader.kt

```diff
@@ -0,0 +1,74 @@
+package org.wordpress.android.ui.sitecreation
+
+import android.support.v4.content.ContextCompat
+import android.support.v7.content.res.AppCompatResources
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.View
+import android.view.ViewGroup
+import android.widget.EditText
+import android.widget.TextView
+import org.wordpress.android.R
+
+class SearchInputWithHeader(rootView: View, onClear: () -> Unit) {
+    private val headerLayout = rootView.findViewById<ViewGroup>(R.id.header_layout)
+    private val headerTitle = rootView.findViewById<TextView>(R.id.title)
+    private val headerSubtitle = rootView.findViewById<TextView>(R.id.subtitle)
+    private val searchInput = rootView.findViewById<EditText>(R.id.input)
+    private val progressBar = rootView.findViewById<View>(R.id.progress_bar)
+    private val clearAllButton = rootView.findViewById<View>(R.id.clear_all_btn)
+
+    var onTextChanged: ((String) -> Unit)? = null
+
+    init {
+        val context = rootView.context
+
+        val drawable = AppCompatResources.getDrawable(context, R.drawable.ic_search_white_24dp)
+        drawable?.setTint(ContextCompat.getColor(context, R.color.grey))
+        searchInput.setCompoundDrawablesRelativeWithIntrinsicBounds(drawable, null, null, null)
+        clearAllButton.background = drawable
+
+        val clearAllLayout = rootView.findViewById<View>(R.id.clear_all_layout)
+        clearAllLayout.setOnClickListener {
+            onClear()
+        }
+
+        searchInput.addTextChangedListener(object : TextWatcher {
+            override fun afterTextChanged(s: Editable?) {}
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                onTextChanged?.invoke(s?.toString() ?: "")
```

Comment:
I'd suggest throwing an error - at least in debug - when the `onTextChanged` is null so we can easily and quickly find the issue.

Reply:
It's possible to use this view without having a callback for the text change, so I don't think the exception is necessary here. Everything will behave the way it should without the `onTextChanged`, so there is nothing to worry about there either. Finally, throwing an exception here forces the setup to be done in a particular way which a component should not do.

If we have to have a valid `onTextChanged`, we need to move it to the constructor which is how I had it before. We needed to flexibility of setting it somewhere like `onViewStateRestored`, so that's why I changed it to this.

Reply:
If it's not clear from my previous comment, if you make that change, set a text, go to the domains screen and go back it'll crash because `onTextChanged` will not be set during restoration.